### PR TITLE
config: Example config file for Creality CR5 Pro HT

### PR DESCRIPTION
**Pull request by bakerbrandond**

### CODE REVIEW

**Comment by bakerbrandond** [comment]
I'll do one more verification tomorrow but otherwise I'm looking forward to feedback.

**Comment by bakerbrandond** [comment]
Ok, I've squashed down my fixups.  I've reviewed the machine description [here](https://www.creality.com/products/creality-cr-5-pro-h-3d-printer) and have ensured the x,y, and z min/max are in safe ranges and verified against the actual machine (needed to shave a bit off of y to be safe).  This is now ready for widespread deployment.

**Comment by bakerbrandond** [comment]
Do note that this PR will result in a printer without functional touchscreen.  How do we handle this?  I see [this](https://klipper.discourse.group/t/dwin-t5uid1-touchscreen-support/2411) outstanding discussion.

**Comment by JamesH1978** [comment]
Thankyou for submitting a PR, please can you refer to https://github.com/Klipper3d/klipper/blob/master/docs/CONTRIBUTING.md paying particular attention to the signed-off-by line.

Also please refer to https://github.com/Klipper3d/klipper/blob/master/docs/Example_Configs.md and make sure you follow the guidelines including adding the regression tests in printers.test as outlined to point 4.

Thanks
James

**Comment by bakerbrandond** [comment]
@JamesH1978 ready for review?

**Review by JamesH1978 — CHANGES_REQUESTED** [review]
Looks fine to me apart from the comment I made about the stowing of the pin.

Can you also confirm that this printer has endstops at axis minimum, and not axis maximum. ie are endstops back right or front left?

Thanks
James

File: config/printer-creality-cr5pro-ht-2022.cfg [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+
+#[display]
+# DWIN screen currently unsupported
+
+[bltouch]
+sensor_pin: ^PD2 #^ar19
+control_pin: PB5 #ar11
+set_output_mode: 5V
+pin_move_time: 0.4
+stow_on_each_sample: False
```
[the remark this answers is not in the document]
Comment (bakerbrandond) [reply config/printer-creality-cr5pro-ht-2022.cfg]: Updated.  Endstops are confirmed front left.

[the remark this answers is not in the document]
Comment (bakerbrandond) [reply config/printer-creality-cr5pro-ht-2022.cfg]: made sure to add Signed-off-by to latest commit.  would you prefer I squash?

**Review by JamesH1978 — APPROVED** [review]
Looks fine to me

Thanks
James

**Comment by KevinOConnor** [comment]
Thanks.

-Kevin

**Comment by hlinden** [comment]
@bakerbrandond Hey, just a quick thank you for this commit. This turned my 1k€ paperweight into an actually usable printer, but I'd never have managed to create a klipper config from scratch.

**Comment by bakerbrandond** [comment]
> @bakerbrandond Hey, just a quick thank you for this commit. This turned my 1k€ paperweight into an actually usable printer, but I'd never have managed to create a klipper config from scratch.

Hey bro! Thanks for taking the time out to share appreciation. Glad I could help. I was able to score one on Craigslist and when I saw there was no config I couldn't stand idly by!

Please let me know how she runs and if you have any issues. As far as I know you and I are the only ones using this printer with Klipper.

Last note... Try OrcaSlicer their built in printer tuning wizards and wiki is great!

**Comment by TheBinaryLoop** [comment]
@bakerbrandond I've got this setup running for a few months now and it works flawlessly. I couldn't get my printer to work with stock settings no matter what I tried. Now I can just press print, go about my day and be sure that it will print correctly (perfect first layer and so on).

**Comment by hlinden** [comment]
@bakerbrandond sorry - I've kept radio silence until now because I still have some issues - make no mistake, Klipper and your config absolutely fixed the printer as far as possible in software. I do have trouble with some filament gliding better or worse in the bowden tube, depending on the bend radius. This means, there is not really a single good value for all things related to handling the filament, when I'm good for printing things on the front left corner I'll get massive underextrusion in the rear right.

That's not really a software problem - while it could somewhat be fixed in software somewhat, probably _(although I don't think Klipper supports position dependend lookup tables for config values)_ I decided to mount a direct extruder, because, mechanical problems should be fixed mechanically first, imho.

The adapter is printed in ASA and lying here, an Orbiter extruder is mounted on top, an adxl345 is stuck to it with some cat 6 soldered neatly on and that's about where the whole thing lost steam mid-december and has been sitting on my workbench since then.

I'll give feedback when that modification is running!

**Comment by bakerbrandond** [comment]
> @bakerbrandond I've got this setup running for a few months now and it works flawlessly. I couldn't get my printer to work with stock settings no matter what I tried. Now I can just press print, go about my day and be sure that it will print correctly (perfect first layer and so on).

So happy to contribute to this community. Happy New Year. Happy Printing!

**Comment by bakerbrandond** [comment]
> @bakerbrandond sorry - I've kept radio silence until now because I still have some issues - make no mistake, Klipper and your config absolutely fixed the printer as far as possible in software. I think I had to adjust the Z-Position of the BLtouch but everything else was really fine out of the box. 
> 
> The rest of my trouble is of mechanical nature: I do have trouble with some filament gliding better or worse in the bowden tube, depending on the bend radius. This means, there is not really a single good value for all things related to handling the filament, when I'm good for printing things on the front left corner I'll get massive underextrusion in the rear right.
> 
> That's not really a software problem - while it could somewhat be fixed in software somewhat, probably _(although I don't think Klipper supports position dependend lookup tables for config values)_ I decided to mount a direct extruder, because, mechanical problems should be fixed mechanically first, imho.
> 
> The adapter is printed in ASA and lying here, an Orbiter extruder is mounted on top, an adxl345 is stuck to it with some cat 6 soldered neatly on and that's about where the whole thing lost steam mid-december and has been sitting on my workbench since then.
> 
> I'll give feedback when that modification is running!

Hmmm that sounds like an interesting solution. I'm wondering if swapping out the bowden tube for a Capricorn would have helped at all.

**Comment by zhujiang0086** [comment]
@bakerbrandond Thank you very much for providing this configuration file. Could you please also share the parameter settings for menuconfig? I am a bit worried about whether I can successfully update the firmware or not.

**Comment by JamesH1978** [comment]
the setting are at the top of the file ```# This file contains common pin mappings for the Creality CR5 Pro HT.
# The mainboard is a Creality 3D v2.5.1 (8-bit mainboard with
# ATMega2560). To use this config, the firmware should be compiled for
# the AVR atmega2560.```

image

**Comment by zhujiang0086** [comment]
@JamesH1978 Thank you! I will try this.

**Comment by DragonSlayerSE** [comment]
Did someone document all the pins on the creality 3d v2.5.1 motherboard? i am trying to run klipper on CR-3040 Pro that has the same mb but not the same wiering
cr3040_pro_mb

**Comment by bakerbrandond** [comment]
> Did someone document all the pins on the creality 3d v2.5.1 motherboard? i am trying to run klipper on CR-3040 Pro that has the same mb but not the same wiering cr3040_pro_mb

what exactly is your issue?

**Pull request merged**